Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGridComboBoxColumn.cs

```diff
@@ -335,7 +335,14 @@ protected override FrameworkElement GenerateElement(DataGridCell cell, object da
                 }
                 else
                 {
-                    textBlockElement.Text = GetDisplayValue(dataItem);
+                    var textBlockBinding = new Binding
+                    {
+                        Source = dataItem,
+                        Path = new PropertyPath(DisplayMemberPath),
+                        Mode = BindingMode.TwoWay
+                    };
+
+                    textBlockElement.SetBinding(TextBlock.TextProperty, textBlockBinding);
```

Comment:
After some more testing I realize that this solution only works if `Binding` is set to the same property as `DisplayMemberPath`. Still struggling to get the path correct here.....

Reply:
@RBrid any suggestions?

Reply:
@filipwa84, I need to know more about your broken scenario. Could you please describe it to me or give me an example? What exactly are you changing programmatically? Maybe provide relevant sample code so we are on the same page. Thank you.

Reply:
@RBird I can post a sample code if you like but it is straight forward really. 

Let's say your combobox column contains an id and another value is shown through DisplayMemberPath, if you go and change the id programmatically the display value isn't updated in the grid. 

Let me know if u need a sample. Can put something together this afternoon if required.

Reply:
@filipwa84 yes a sample would be helpful, also attaching it to the issue and linking here would be great.  Thanks!

Reply:
@filipwa84, thanks for the info. Let me try to repro this on my side, and think about how I would address this.

Reply:
All right @filipwa84, I have investigated this and came up with the following solution for DataGridComboBoxColumn.cs:

Changes from scratch:

- Add `using System.Collections.Generic;` between `using System.Collections;` and `using System.Collections.Specialized;`

- Add         `private HashSet<object> _notifyingDataItems;`
below         `private Brush _foreground;`

- Add HookDataItemPropertyChanged call where you had your fix:
```
            if (Binding != null && EnsureOwningGrid())
            {
                if (string.IsNullOrEmpty(DisplayMemberPath))
                {
                    textBlockElement.SetBinding(TextBlock.TextProperty, Binding);
                }
                else
                {
                    textBlockElement.Text = GetDisplayValue(dataItem);

                    HookDataItemPropertyChanged(dataItem);
                }
            }
```
- Add `_owningGrid.UnloadingRow += OwningGrid_UnloadingRow;`
below `_owningGrid.LoadingRow += OwningGrid_LoadingRow;`

- Update OwningGrid_LoadingRow:
```
        private void OwningGrid_LoadingRow(object sender, DataGridRowEventArgs e)
        {
            HookDataItemPropertyChanged(e.Row.DataContext);
            SetDisplayMemberPathValue(e.Row);
        }
```

- Add 
```
        private void OwningGrid_UnloadingRow(object sender, DataGridRowEventArgs e)
        {
            UnhookDataItemPropertyChanged(e.Row.DataContext);
        }
```

- Update Columns_CollectionChanged:
```
        private void Columns_CollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
        {
            if (OwningGrid == null && _owningGrid != null)
            {
                _notifyingDataItems.Clear();
                _owningGrid.Columns.CollectionChanged -= new NotifyCollectionChangedEventHandler(Columns_CollectionChanged);
                _owningGrid.LoadingRow -= OwningGrid_LoadingRow;
                _owningGrid.UnloadingRow -= OwningGrid_UnloadingRow;
                _owningGrid.CellEditEnded -= OwningGrid_CellEditEnded;
                _owningGrid = null;
            }
        }
```

- Add
```
        private void HookDataItemPropertyChanged(object dataItem)
        {
            if (Binding.Mode == BindingMode.OneTime)
            {
                return;
            }

            var notifyingDataItem = dataItem as INotifyPropertyChanged;

            if (notifyingDataItem == null)
            {
                return;
            }

            if (_notifyingDataItems == null)
            {
                _notifyingDataItems = new HashSet<object>();
            }

            if (!_notifyingDataItems.Contains(dataItem))
            {
                notifyingDataItem.PropertyChanged += DataItem_PropertyChanged;
                _notifyingDataItems.Add(dataItem);
            }
        }

        private void UnhookDataItemPropertyChanged(object dataItem)
        {
            if (_notifyingDataItems == null)
            {
                return;
            }

            var notifyingDataItem = dataItem as INotifyPropertyChanged;

            if (notifyingDataItem == null)
            {
                return;
            }

            if (_notifyingDataItems.Contains(dataItem))
            {
                notifyingDataItem.PropertyChanged -= DataItem_PropertyChanged;
                _notifyingDataItems.Remove(dataItem);
            }
        }

        private void DataItem_PropertyChanged(object dataItem, PropertyChangedEventArgs e)
        {
            if (OwningGrid != null &&
                Binding != null &&
                Binding.Path != null &&
                Binding.Path.Path == e.PropertyName)
            {
                DataGridRow dataGridRow = OwningGrid.GetRowFromItem(dataItem);

                if (dataGridRow != null)
                {
                    var textBlockElement = GetCellContent(dataGridRow) as TextBlock;

                    if (textBlockElement != null)
                    {
                        textBlockElement.Text = GetDisplayValue(dataItem);
                    }
                }
            }
        }
```

This seems to work fine for me. Could you please give it a try?  Thank you.

Reply:
@RBrid That's an elegant solution! One thing though, where we have `Binding.Mode != Mode.OneTime` we should actually check the Binding of the column's ItemsSource right? To my understanding Binding is always TwoWay.

Reply:
@RBrid Did you have a chance to look at it?

Reply:
@michael-hawker and @RBrid Have you had a chance to look at the changes yet?

Reply:
@filipwa84, sorry for the long delay.   I am not sure I understand your question. Are you talking about line 53 in DataGridBoundColumn.cs, `_binding.Mode = BindingMode.TwoWay;`?

DataGridBoundColumn.Binding is initially set to TwoWay. But the app developer can always overwrite it later and set it to OneTime.   If I remember correctly, that is why I had the check

```
            if (Binding.Mode == BindingMode.OneTime)
            {
                return;
            }

```
Is that what you meant?   Thank you!

Reply:
I was referring to this section:
```
if (Binding.Mode == BindingMode.OneTime)
{
                return;
}
```
My understanding was that Binding would always be TwoWay. Thats why I asked if we shouldn't check the binding of the ItemsSource instead. As you point out it can be changed, checking Binding makes more sense. Sorry for the confusion.